addons/web/static/src/webclient/company_service.js

```diff
@@ -51,9 +51,39 @@ function getCompanyIdsFromBrowser(hash) {
     return cids || [];
 }
 
+const errorHandlerRegistry = registry.category("error_handlers");
+function accessErrorHandler(env, error, originalError) {
+    const router = env.services.router;
+    const hash = router.current.hash;
+    if (!hash._company_switching) {
+        return false;
+    }
+    if (originalError?.exceptionName === "odoo.exceptions.AccessError") {
+        const { model, id, view_type } = hash;
```

Comment:
I was thinking about that earlier today and didn't have the chance to tell you about it... Are the model and id you don't have access to specified somewhere in the error? If so, it would be nice to check that they are the same as in the url...

Reply:
As I thought, the error doesn't contain raw data or easily parsable and robust text